pkg/asset/installconfig/openstack/validation/cloudinfo.go

```diff
@@ -374,6 +383,39 @@ func getComputeLimits(opts *clientconfig.ClientOpts, projectID string) ([]record
 	return records, nil
 }
 
+func getNetworkLimits(opts *clientconfig.ClientOpts, projectID string) ([]record, error) {
+	networkClient, err := clientconfig.NewServiceClient("network", opts)
```

Comment:
Note that this API hits keystone every time. If you get an AuthenticatedClient and use it to instantiate a service client it doesn't have to hit keystone. That's something we should look at generally, though.

Reply:
Don't we call ```getNetworkLimits``` only once?

Reply:
Probably, but we hit Keystone every time we create a service client. i.e. We hit it once for Nova and once again for Neutron. If we used AuthenticatedClient (or whatever it's called) we'd only hit it once, and reuse that token for all service clients.

Reply:
I agree with you. However I was told to use ``NewServiceClient`` because it's an abstraction from the actual client function, e.g. ```openstack.NewNetworkV2```.
By using ``clientconfig.NewServiceClient("network", opts)``, we don't care if the default for Network client switches to NewNetworkV3, it'll be transparent for us.

I'll defer to @mandre on this one, I'm happy to call ```openstack.NewNetworkV2``` (and change compute as well).

Reply:
Agreed we should re-use the same `AuthenticatedClient` for all service clients. This is however out of scope for this change.

For now please make `loadQuotas()` a method of `CloudInfo` and use `ci.clients.networkClient`.

Reply:
ack, will close that comment then and we'll fix it separately.

Reply:
and ack, I'll make `loadQuotas()` a method of `CloudInfo` and use `ci.clients.networkClient`.

Reply:
> Agreed we should re-use the same `AuthenticatedClient` for all service clients. This is however out of scope for this change.
> 
> For now please make `loadQuotas()` a method of `CloudInfo` and use `ci.clients.networkClient`.

Please take a look at https://github.com/gophercloud/utils/blob/master/openstack/clientconfig/requests.go#L748-L851
I actually thing it's good to use `NewServiceClient()` since there is a bunch of logic for variables and SSL stuff.

Reply:
We're using `NewServiceClient()` to allocate `ci.clients.networkClient`:
https://github.com/openshift/installer/blob/338310d4f7a3383054cb12c9dd19df258c313640/pkg/asset/installconfig/openstack/validation/cloudinfo.go#L79
So it's there to use freely if we make `loadQuotas()`, `getNetworkLimits()`, etc,  methods of `CloudInfo`.

Reply:
ok I see what you mean now. I'll take a look and update accordingly.

Reply:
done, and I fixed computed too, in another commit!